### DISCUSSION TITLE
How to make `vercel dev` work with `venv` on python serverless functions

### DISCUSSION THREAD

**Discussion by cohlar** [vercel/vercel#11369] [Help]
### Question

Hi,

I am trying to make `vercel dev` work with a python serverless function, where dependencies are installed on a virtual environment. Unfortunately so far I could not make it work - everything works perfectly if I install dependencies globally on my machine.

Here is a [minimum reproduction repo](https://github.com/cohlar/vercel_python_api_sample), and below is what I have tried.

**What doesn't work:**
```
python -m venv .venv
.venv/Scripts/activate
python -m pip install -r requirements.txt
vercel dev
```

When calling the `/api` endpoint, I get the following python exception: `ModuleNotFoundError: No module named 'pandas'`.

**What works:**
```
python -m pip install -r requirements.txt
vercel dev
```

That's not ideal because I want to have a separate version of python (Vercel seems to support python 3.9 as of now) and dedicated requirement versions for this project.

Is there maybe a way to specify to `vercel` to use the virtual enviroment for the repo?

Thanks for your help.

**Comment by vsimplify** [comment]
hi, how did you fix this problem?
please share

**Reply by WhiteMinds** [reply]
My solution is `pipenv run vercel dev`.
